### PR TITLE
James 3805 fix PulsarMailQueueTest.dequeueShouldBeConcurrent and a few other unstability

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by chibenwa** [comment]
Fails on pulsar:

```
org.apache.pulsar.client.admin.PulsarAdminException$NotFoundException: Tenant does not exist
Caused by: org.apache.pulsar.shade.javax.ws.rs.NotFoundException: HTTP 404 Tenant does not exist
```



  | Test Result (11 failures  / +11)org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getMailQueueShouldReturnMailQueueIfCreatedorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getUsedMailQueuesShouldReturnWhenNoMailQueueorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getMailQueueShouldReturnEmptyIfNotCreatedorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getUsedMailQueuesShouldNotReturnDuplicateorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getMailQueueShouldNotReturnTheSameQueueForTwoDifferentNamesorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getUsedMailQueuesShouldReturnPreviouslyCreatedMailQueuesorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getMailQueueShouldReturnEmptyIfQueueDoesNotExistorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.createMailQueueShouldNotConflictIfAlreadyExistsorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.createAlreadyCreatedQueueShouldReturnPreviouslyCreatedMailQueueInstanceorg.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getExistingMailQueueShouldReturnPreviouslyCreatedMailQueueInstance
-- | --


[Test Result](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/) (11 failures / +11)

    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getMailQueueShouldReturnMailQueueIfCreated](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/getMailQueueShouldReturnMailQueueIfCreated/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getUsedMailQueuesShouldReturnWhenNoMailQueue](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/getUsedMailQueuesShouldReturnWhenNoMailQueue/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getMailQueueShouldReturnEmptyIfNotCreated](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/getMailQueueShouldReturnEmptyIfNotCreated/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getUsedMailQueuesShouldNotReturnDuplicate](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/getUsedMailQueuesShouldNotReturnDuplicate/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getMailQueueShouldNotReturnTheSameQueueForTwoDifferentNames](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/getMailQueueShouldNotReturnTheSameQueueForTwoDifferentNames/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getUsedMailQueuesShouldReturnPreviouslyCreatedMailQueues](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/getUsedMailQueuesShouldReturnPreviouslyCreatedMailQueues/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getMailQueueShouldReturnEmptyIfQueueDoesNotExist](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/getMailQueueShouldReturnEmptyIfQueueDoesNotExist/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.createMailQueueShouldNotConflictIfAlreadyExists](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/createMailQueueShouldNotConflictIfAlreadyExists/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.createAlreadyCreatedQueueShouldReturnPreviouslyCreatedMailQueueInstance](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/createAlreadyCreatedQueueShouldReturnPreviouslyCreatedMailQueueInstance/)
    [org.apache.james.queue.pulsar.PulsarMailQueueFactoryTest.getExistingMailQueueShouldReturnPreviouslyCreatedMailQueueInstance](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1190/2/testReport/junit/org.apache.james.queue.pulsar/PulsarMailQueueFactoryTest/getExistingMailQueueShouldReturnPreviouslyCreatedMailQueueInstance/)
    
Re-launching the build...

**Comment by jeantil** [comment]
So I'm at 70 runs of `PulsarMailqueueFactoryTest` and it doesn't seem to want to fail, I'll assume a CI fluke and try `PulsarMailQueueTest#removeByRecipientShouldRemoveSpecificEmailWhenMultipleRecipients`

File: server/queue/queue-pulsar/src/main/scala/org/apache/james/queue/pulsar/PulsarMailQueueConfiguration.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.apache.james.queue.pulsar
```
Comment (Arsnael) [inline server/queue/queue-pulsar/src/main/scala/org/apache/james/queue/pulsar/PulsarMailQueueConfiguration.scala:1]: License missing

**Comment by jeantil** [comment]
:green_apple: build, I'll rebase and merge

**Pull request merged**